pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/httproute_reconciler.go

```diff
@@ -595,38 +595,57 @@ func (r *RawHTTPRouteReconciler) reconcileTransformerHTTPRoute(ctx context.Conte
 	if err != nil {
 		return err
 	}
+
+	// reconcile httpRoute
+	httpRouteName := constants.TransformerServiceName(isvc.Name)
+	existingHttpRoute := &gatewayapiv1.HTTPRoute{}
+	getExistingErr := r.client.Get(ctx, types.NamespacedName{
+		Name: httpRouteName, Namespace: isvc.Namespace,
+	}, existingHttpRoute)
+	httpRouteIsNotFound := apierr.IsNotFound(getExistingErr)
+	if getExistingErr != nil && !httpRouteIsNotFound {
+		return fmt.Errorf("failed to get existing http route: %w", getExistingErr)
+	}
+
+	// ISVC is stopped, delete the httproute if it exists, otherwise, do nothing
+	forceStopRuntime := utils.GetForceStopRuntime(isvc)
+	if (getExistingErr != nil && httpRouteIsNotFound) && forceStopRuntime {
+		return nil
+	}
+	if forceStopRuntime {
+		if ctrl := metav1.GetControllerOf(existingHttpRoute); ctrl != nil && ctrl.UID == isvc.UID {
+			log.Info("The InferenceService is marked as stopped — deleting its associated http route", "name", httpRouteName)
+			if err := r.client.Delete(ctx, existingHttpRoute); err != nil {
+				return fmt.Errorf("failed to delete HTTPRoute %s/%s: %w", desired.Namespace, desired.Name, err)
+			}
+		}
+		return nil
+	}
+
+	// Create or update the httproute to match the desired state
 	if desired == nil {
 		return nil
 	}
 	if err := controllerutil.SetControllerReference(isvc, desired, r.scheme); err != nil {
 		log.Error(err, "Failed to set controller reference for transformer HttpRoute", "name", desired.Name)
+		return err
 	}
-	existing := &gatewayapiv1.HTTPRoute{}
-	err = r.client.Get(ctx, types.NamespacedName{Name: desired.Name, Namespace: isvc.Namespace}, existing)
-	if err != nil {
-		if apierr.IsNotFound(err) {
-			log.Info("Creating transformer HttpRoute resource", "name", desired.Name)
-			if err := r.client.Create(ctx, desired); err != nil {
-				log.Error(err, "Failed to create transformer HttpRoute", "name", desired.Name)
-				return err
-			}
-		} else {
+
+	if getExistingErr != nil && httpRouteIsNotFound {
+		log.Info("Creating transformer HttpRoute resource", "name", desired.Name)
+		if err := r.client.Create(ctx, desired); err != nil {
+			log.Error(err, "Failed to create transformer HttpRoute", "name", desired.Name)
 			return err
 		}
-	} else {
-		// Set ResourceVersion which is required for update operation.
-		desired.ResourceVersion = existing.ResourceVersion
-		// Do a dry-run update to avoid diffs generated by default values.
-		// This will populate our local httpRoute with any default values that are present on the remote version.
-		if err := r.client.Update(ctx, desired, client.DryRunAll); err != nil {
-			log.Error(err, "Failed to perform dry-run update for transformer HttpRoute", "name", desired.Name)
+		return nil
+	}
+	// Set ResourceVersion which is required for update operation.
+	desired.ResourceVersion = existingHttpRoute.ResourceVersion
+	if !semanticHttpRouteEquals(desired, existingHttpRoute) {
+		if err := r.client.Update(ctx, desired); err != nil {
```

Comment:
The original logic was generally used to compare objects before updating, but I noticed it was removed.
Can you clarify why this change was made?

Reply:
@pierDipi suggested the change - it is paired with changing the semanticHttpRouteEquals to use DeepDerivative (see https://github.com/kserve/kserve/pull/4455#discussion_r2136358947)

Reply:
have you seen this comment? https://github.com/kserve/kserve/pull/4455/files#r2141884761